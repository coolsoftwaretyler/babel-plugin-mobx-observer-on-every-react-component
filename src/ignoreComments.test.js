import { expect, test, describe } from "bun:test";
import { transform } from "@babel/core";
import pluginSyntaxJsx from "@babel/plugin-syntax-jsx";
import pluginTransReactJsx from "@babel/plugin-transform-react-jsx";
import pluginSyntaxDecorators from "@babel/plugin-syntax-decorators";
import pluginSyntaxDecoratorsLegacy from "@babel/plugin-proposal-decorators";
import autoObserverPlugin from "./index.ts";

const runTransform = (input) => {
    return transform(input, {
        plugins: [pluginSyntaxJsx, pluginTransReactJsx, [pluginSyntaxDecorators, { version: "2018-09", decoratorsBeforeExport: false }], autoObserverPlugin],
    });
};

const runTransformLegacy = (input) => {
    return transform(input, {
        plugins: [pluginSyntaxJsx, pluginTransReactJsx, [pluginSyntaxDecoratorsLegacy, { legacy: true }], autoObserverPlugin],
    });
};

describe("Mobx Observer Babel Plugin", () => {
    describe("ignoreComments", () => {
        describe("when the comment is to ignore the full file", () => {
            test("ignores everything in the file", () => {
                const source = `
          // @auto-observer-ignore-file
          class MyComponentClassDeclaration { render() { return <div>Hello World</div>; } }
          const MyArrowObserver = () => <div>Hello World</div>
          const MyComponentClassExpression = class MyComponent { render() { return <div>Hello World</div>; } }
          function MyComponentDeclaration() { return <div>Hello World</div>; }
          const MyComponentExpression = function MyComponentExpression() { return <div>Hello World</div>; }
          `;
                const out = runTransform(source);

                expect(out.code).toMatchSnapshot();
            });
        });
        describe("when the comment is to ignore a specific block", () => {
            test("ignores the specific block", () => {
                const source = `
        // @auto-observer-ignore-block
        class MyComponentClassDeclaration { render() { return <div>Hello World</div>; } }
        const MyArrowObserver = () => <div>Hello World</div>
        const MyComponentClassExpression = class MyComponent { render() { return <div>Hello World</div>; } }
        function MyComponentDeclaration() { return <div>Hello World</div>; }
        const MyComponentExpression = function MyComponentExpression() { return <div>Hello World</div>; }
        `;
                const out = runTransform(source);

                expect(out.code).toMatchSnapshot();
            });
        });
    });
});
