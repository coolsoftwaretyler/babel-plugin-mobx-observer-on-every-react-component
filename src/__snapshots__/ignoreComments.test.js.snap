// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx Observer Babel Plugin ignoreComments when the comment is to ignore the full file ignores everything in the file 1`] = `
"// @auto-observer-ignore-file
class MyComponentClassDeclaration {
  render() {
    return /*#__PURE__*/React.createElement("div", null, "Hello World");
  }
}
const MyArrowObserver = () => /*#__PURE__*/React.createElement("div", null, "Hello World");
const MyComponentClassExpression = class MyComponent {
  render() {
    return /*#__PURE__*/React.createElement("div", null, "Hello World");
  }
};
function MyComponentDeclaration() {
  return /*#__PURE__*/React.createElement("div", null, "Hello World");
}
const MyComponentExpression = function MyComponentExpression() {
  return /*#__PURE__*/React.createElement("div", null, "Hello World");
};"
`;

exports[`Mobx Observer Babel Plugin ignoreComments when the comment is to ignore a specific block ignores the specific block 1`] = `
"import { observer } from "mobx-react";
// @auto-observer-ignore-block
class MyComponentClassDeclaration {
  render() {
    return /*#__PURE__*/React.createElement("div", null, "Hello World");
  }
}
const MyArrowObserver = observer(() => /*#__PURE__*/React.createElement("div", null, "Hello World"));
const MyComponentClassExpression = observer(class MyComponent {
  render() {
    return /*#__PURE__*/React.createElement("div", null, "Hello World");
  }
});
const MyComponentDeclaration = observer(function MyComponentDeclaration() {
  return /*#__PURE__*/React.createElement("div", null, "Hello World");
});
const MyComponentExpression = observer(function MyComponentExpression() {
  return /*#__PURE__*/React.createElement("div", null, "Hello World");
});"
`;
